### DISCUSSION TITLE
Problem with a specific IBAN

### DISCUSSION THREAD

**Discussion by povallat** [manuelbl/SwissQRBill.NET#64] [Q&A]
Hello,
I'm using SwissQRBill.NET and it's really a top development! Thanks a lot!
I have a problem with a specific IBAN (the other IBANs work fine). This IBAN is [CH06 3080 8002 9858 5042 6]. As for the other IBANs, I remove the spaces, which gives me [CH0630808002985850426]. I then create my bill with Bill bill = new Bill {...} with the necessary parameters including Account = "CH0630808002985850426". The Payments.IsValidIban(iban) function tells me that the iban is in ok. When I launch generation with QRBill.Generate(bill), an exception is thrown and bill takes the value null. If I keep the other data of my bill (address, name...) and enter another valid IBAN, everything works fine.

Can you help? 

Thanks in advance and best regards, 

Exception:
Codecrete.SwissQRBill.Generator.QRBillValidationException
  HResult=0x80131500
  Message=QR bill data is invalid
  Source=Codecrete.SwissQRBill.Core
  Arborescence des appels de procédure :
   à Codecrete.SwissQRBill.Generator.QRBill.ValidateAndGenerate(Bill bill, ICanvas canvas)
   à Codecrete.SwissQRBill.Generator.QRBill.Generate(Bill bill)
   à QrCodeFacturesPov.FQrCodeFacturesPov.createBillPdfToolStripMenuItem_Click(Object sender, EventArgs e) dans D:\source\repos\QRCodeFactures\QrCodeFacturesPov\FQrCodeFacturesPov.cs :ligne 669
   à System.Windows.Forms.ToolStripItem.RaiseEvent(Object key, EventArgs e)
   à System.Windows.Forms.ToolStripMenuItem.OnClick(EventArgs e)
   à System.Windows.Forms.ToolStripItem.HandleClick(EventArgs e)
   à System.Windows.Forms.ToolStripItem.HandleMouseUp(MouseEventArgs e)
   à System.Windows.Forms.ToolStripItem.FireEventInteractive(EventArgs e, ToolStripItemEventType met)
   à System.Windows.Forms.ToolStripItem.FireEvent(EventArgs e, ToolStripItemEventType met)
   à System.Windows.Forms.ToolStrip.OnMouseUp(MouseEventArgs mea)
   à System.Windows.Forms.ToolStripDropDown.OnMouseUp(MouseEventArgs mea)
   à System.Windows.Forms.Control.WmMouseUp(Message& m, MouseButtons button, Int32 clicks)
   à System.Windows.Forms.Control.WndProc(Message& m)
   à System.Windows.Forms.ScrollableControl.WndProc(Message& m)
   à System.Windows.Forms.ToolStrip.WndProc(Message& m)
   à System.Windows.Forms.ToolStripDropDown.WndProc(Message& m)
   à System.Windows.Forms.Control.ControlNativeWindow.OnMessage(Message& m)
   à System.Windows.Forms.Control.ControlNativeWindow.WndProc(Message& m)
   à System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, WM msg, IntPtr wparam, IntPtr lparam)

**Comment by manuelbl** [comment]
The thrown exception contains validation messages with more detail information about the problem. 

Another option is to enter the bill data in https://www.codecrete.net/qrbill/. It will also provide the details. 

The shown IBAN is a QR-IBAN. It requires that the bill uses a numeric reference. The most likely cause is that your bill has no reference or an alphanumeric one.

**Comment by povallat** [comment]
Hello,

 

Many thanks for your quick help. I've found my problem: the IBAN in the fourth and fifth positions is 31, so I need a reference number. 

The link https://www.codecrete.net/qrbill/ helped me find this problem.

 

Thanks again and, if you're interested, I'll be happy to share my development with you.

 

POV

 

De : Manuel Bl. ***@***.***> 
Envoyé : mercredi, 21 juin 2023 16:45
À : manuelbl/SwissQRBill.NET ***@***.***>
Cc : povallat ***@***.***>; Author ***@***.***>
Objet : Re: [manuelbl/SwissQRBill.NET] Problem with a specific IBAN (Discussion #64)

 

The thrown exception contains validation messages with more detail information about the problem.

Another option is to enter the bill data in https://www.codecrete.net/qrbill/. It will also provide the details.

The shown IBAN is a QR-IBAN. It requires that the bill uses a numeric reference. The most likely cause is that your bill has no reference or an alphanumeric one.
